### PR TITLE
[infra] Upload latest.version metadata file for most recent build.

### PR DESCRIPTION
**Pull request by Dor1s**
Draft for now, I'll probably move the curl step into a helper function (to avoid duplication with the coverage job).

### CODE REVIEW

**Comment by Dor1s** [comment]
https://github.com/google/oss-fuzz/issues/3378

File: infra/gcb/build_project.py [hunk 1]
```diff
@@ -206,8 +215,8 @@ def get_build_steps(project_dir):
                     # `cd /src && cd {workdir}` (where {workdir} is parsed from
                     # the Dockerfile). Container Builder overrides our workdir
                     # so we need to add this step to set it back.
-                    ('rm -r /out && cd /src && cd {workdir} && mkdir -p {out} && '
```
Comment (jonathanmetzman) [inline infra/gcb/build_project.py]: why this change?

File: infra/gcb/build_project.py [hunk 2]
```diff
@@ -324,6 +333,9 @@ def get_build_steps(project_dir):
                     targets_list_url,
                 ],
             },
+            # upload the latest.version file
```
Comment (jonathanmetzman) [inline infra/gcb/build_project.py]: nit: Capitalize comment and end with period.

**Comment by Dor1s** [comment]
Tested on yara build, it worked:

```bash
$ gsutil ls gs://clusterfuzz-builds/yara/*-latest.version
gs://clusterfuzz-builds/yara/yara-address-latest.version
gs://clusterfuzz-builds/yara/yara-memory-latest.version
gs://clusterfuzz-builds/yara/yara-undefined-latest.version
```

File: infra/gcb/build_project.py (hunk 1 above)
Comment (Dor1s) [reply infra/gcb/build_project.py]: linter complained, looks like some of the previous formatting changes increased the indentation here and the line became too long

File: infra/gcb/build_project.py (hunk 2 above)
Comment (Dor1s) [reply infra/gcb/build_project.py]: consistency with the other comments!!! (sorry I've clicked merged literally the same second your comments popped up)
